### PR TITLE
Shuffle RPC packages

### PR DESCRIPTION
**Pull request by roman-khimov**
### Problem

Step 1 of #2597.

### Solution

Differs a bit from what was planned, but I think it's better this way. `neorpc/result` is still kept, I'm still not so sure about it. But we have Blockchainer finally gone as a bonus.

### CODE REVIEW

File: cli/server/server.go [hunk 1]
```diff
@@ -21,10 +21,10 @@ import (
 	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
 	"github.com/nspcc-dev/neo-go/pkg/io"
 	"github.com/nspcc-dev/neo-go/pkg/network"
-	"github.com/nspcc-dev/neo-go/pkg/network/metrics"
-	"github.com/nspcc-dev/neo-go/pkg/services/rpcsrv"
+	"github.com/nspcc-dev/neo-go/pkg/services/metrics"
 	"github.com/nspcc-dev/neo-go/pkg/services/notary"
 	"github.com/nspcc-dev/neo-go/pkg/services/oracle"
+	"github.com/nspcc-dev/neo-go/pkg/services/rpcsrv"
```
Comment (fyrchik) [inline cli/server/server.go:27]: Probably, a part of the first commit.

File: pkg/neorpc/types.go [hunk 2]
```diff
@@ -17,10 +22,10 @@ const (
 )
 
 type (
-	// Raw represents JSON-RPC request. It's generic enough to be used in many
+	// Request represents JSON-RPC request. It's generic enough to be used in many
```
Comment (fyrchik) [inline pkg/neorpc/types.go:25]: Worth adding `request` to the commit message.

File: pkg/services/rpcsrv/server.go [hunk 3]
```diff
@@ -196,7 +248,7 @@ var invalidBlockHeightError = func(index int, height int) *neorpc.Error {
 var upgrader = websocket.Upgrader{}
 
 // New creates a new Server struct.
-func New(chain blockchainer.Blockchainer, conf config.RPC, coreServer *network.Server,
+func New(chain Ledger, conf config.RPC, coreServer *network.Server,
```
Comment (fyrchik) [inline pkg/services/rpcsrv/server.go:250]: We already have a dependency on `core`, do we need this interface at all? It seems all callsites use `core.Blockchain`.

**Review by fyrchik — APPROVED** [review]

File: pkg/services/rpcsrv/server.go (hunk 3 above)
Comment (roman-khimov) [reply pkg/services/rpcsrv/server.go:250]: I hope we can remove it in the future. That's the model we have for all of the other services and RPC shouldn't be different.

File: cli/server/server.go (hunk 1 above)
Comment (roman-khimov) [reply cli/server/server.go:27]: Actually, it's just an automatic `fmt`. `rpcsrv` dependcy itself was properly introduced in the first commit.

File: pkg/neorpc/types.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/neorpc/types.go:25]: Fixed.

**Pull request merged**